pyk/src/pyk/kcfg/show.py

```diff
@@ -30,31 +32,26 @@
     from ..cterm import CSubst
     from ..kast import KInner
     from ..kast.outer import KDefinition, KFlatModule, KImport, KSentence
-    from ..ktool.kprint import KPrint
     from .kcfg import NodeIdLike
 
 _LOGGER: Final = logging.getLogger(__name__)
 
 
 class NodePrinter:
-    kprint: KPrint
+    cterm_show: CTermShow
```

Comment:
Target for future refactorings: this class should be made abstract, as what details I want to show for a node varies from definition to definition (and even for a given definition there might be more ythan a single useful views). Displaying a `CTerm` as in this current implementation is a reasonable default implementation.

Also, the abstract interface should probably define separate abstract functions for providing a list of labels that should be shown in the tree, and for displaying the actual node content.

Reply:
The next refactoring does remove the actual node content display (which is fully moved into `CTermShow`). I was hoping to entirely remove the `NodePrinter` hierarchy, but it's somewhat complicated by the fact that `KCFGShow.pretty_segments` and `APRProofShow.pretty_segments` both take a `KCFG` and need to print node attrs, but we want to print different node attrs depending on if it's a `KCFG` or an `APRProof`. So I'm a bit stuck there, but at least we can totally lift the printing out of `NodePrinter` later (and I'll get that change in once this PR is propagated downstream).